### PR TITLE
UPSTREAM: <carry>: eliminate unnecessary closure in openshift configu…

### PR DESCRIPTION
**Pull request by deads2k**
…ration wiring

This eliminates indirection through a global and instead directly wires the globals.  It's still ugly and order dependent, but one layer of indirection is gone.

### CODE REVIEW

File: vendor/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -117,17 +117,15 @@ cluster's shared state through which all other components interact.`,
 			verflag.PrintAndExitIfRequested()
 
 			if len(s.OpenShiftConfig) > 0 {
+				// if we are running openshift, we modify the admission chain defaults accordingly
+				admissionenablement.InstallOpenShiftAdmissionPlugins(s)
```
Comment (stlaz) [inline vendor/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go:121]: no action needed, just curious: the patching function assignment was noop here, just a side effect elsewhere, so is there anything to this admission plugins install move?

Comment (deads2k) [reply vendor/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go:121]: > no action needed, just curious: the patching function assignment was noop here, just a side effect elsewhere, so is there anything to this admission plugins install move?

doing it early made it easier for me to read and reason about. nothing special.

**Comment by sttts** [comment]
Sgtm
/approve
/assign @marun

**Comment by marun** [comment]
/lgtm

@deads2k I'm assuming this can be squashed on rebase?

**Pull request merged**